File: 0045-jump-game-ii/0045-jump-game-ii.go
func jump(nums []int) int {
    n := len(nums)
    last, maxi, ans := nums[0], nums[0], 0
    for i := 1; i < n; {
        j := i
        for j < n && j <= last {
            maxi = int(math.Max(float64(maxi), float64(j + nums[j])))
            j++
        }
        i = j
        ans++
        last = maxi
    }
    return ans
}

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int last = nums[0], maxi = nums[0], ans = 0;
        for(int i = 1; i < n;) {
            int j = i;
            while(j < n && j <= last) {
                maxi = max(maxi, j + nums[j]);
                j++;
            }
            i = j;
            ans++;
            last = maxi;
        }
        return ans;
    }
};